File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.py
class Solution:
    def peakIndexInMountainArray(self, arr: list[int]) -> int:
        s=1
        e=len(arr)-2
        while(s<=e):
            mid=(s+e)//2
            if(arr[mid-1]<arr[mid] and arr[mid]>arr[mid+1]):
                return mid
            elif(arr[mid-1]>arr[mid]):
                e=mid-1
            else:
                s=mid+1
        

File: 0852-peak-index-in-a-mountain-array/0852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int s=1;
        int e=arr.length-1;
        while(s<=e)
        {
            int mid=(s+e)/2;
            if(arr[mid-1]<arr[mid] && arr[mid]>arr[mid+1])
            {
                return mid;
            }
            else if(arr[mid-1]>arr[mid])
            {
                e=mid-1;
            }
            else
            {
                s=mid+1;
            }
        }
        
     return -1;   
    }
}